libraries/wallet/wallet.cpp

```diff
@@ -1755,41 +1802,37 @@ fc::ecc::private_key wallet_api::derive_private_key(const std::string& prefix_st
 
         annotated_signed_transaction wallet_api::update_account_auth_threshold( string account_name, authority_type type, uint32_t threshold, bool broadcast )
         {
-            FC_ASSERT( !is_locked() );
+            WALLET_CHECK_UNLOCKED();
 
-            auto accounts = my->_remote_database_api->get_accounts( { account_name } );
-            FC_ASSERT( accounts.size() == 1, "Account does not exist" );
-            FC_ASSERT( account_name == accounts[0].name, "Account name doesn't match?" );
-            FC_ASSERT( threshold != 0, "Authority is implicitly satisfied" );
+            auto account = get_account(account_name);
+            GOLOS_CHECK_PARAM(threshold, GOLOS_CHECK_VALUE(threshold != 0, "Authority is implicitly satisfied"));
```

Comment:
Moving this check before `get_account` will remove unnecessary remote request in case of bad `threshold`

Reply:
Fixed.